pkg/kubecache/meta/informers_init.go

```diff
@@ -263,10 +268,20 @@ func (inf *Informers) initPodInformer(informerFactory informers.SharedInformerFa
 	return nil
 }
 
-func envToMap(env []v1.EnvVar) map[string]string {
+func envToMap(kc kubernetes.Interface, objMeta metav1.ObjectMeta, c *v1.Container) map[string]string {
 	envMap := map[string]string{}
-	for _, envV := range env {
-		envMap[envV.Name] = envV.Value
+	for _, envV := range c.Env {
+		if _, ok := usefulEnvVars[envV.Name]; ok {
```

Comment:
To save on memory for containers which have a lot of environment variables, we specifically look for the two we care about.